src/base.jl

```diff
@@ -60,24 +60,37 @@ System(; kwargs...)
     containing subtypes of `Component`.
 
 # Keyword arguments
-- `ext::Dict`: Contains user-defined parameters. Should only contain standard types.
-- `frequency::Float64`: (default = 60.0) Operating frequency (Hz)
+- `name::String`: System name.
+- `description::String`: System description.
+- `frequency::Float64`: (default = 60.0) Operating frequency (Hz).
 - `runchecks::Bool`: Run available checks on input fields and when add_component! is called.
   Throws InvalidValue if an error is found.
+- `generator_mapping`: A dictionary mapping generator names to their corresponding topologies. This is used to associate generators with their respective buses when parsing from CSV.
 - `time_series_in_memory::Bool=false`: Store time series data in memory instead of HDF5.
 - `time_series_directory::Union{Nothing, String}`: Directory for the time series HDF5 file.
-    Defaults to the tmp file system
+    Defaults to the tmp file system.
+- `timeseries_metadata_file`: Path to a file containing time series metadata descriptors. This is used to add time series data to the system from files.
```

Comment:
Adding documentation of these parameters is nice. However, it also adds potential for confusion. How does a user know which kwarg goes with which constructor? Take `timeseries_metadata_file`, for example. It is only accepted (or processed) in one System constructor in power_system_table_data.jl. The user will only know it if they read that function signature. Should we separate the kwargs that apply to all (e.g., time_series_read_only) from constructor-specific kwargs? Maybe the problem is that we need to name the constructors differently (from_psse, from_table_data, from_dyr, etc.)